src/connection-manager/topology.js

```diff
@@ -0,0 +1,102 @@
+'use strict'
+
+class Topology {
+  /**
+   * @param {Object} props
+   * @param {number} props.min minimum needed connections (default: 0)
+   * @param {number} props.max maximum needed connections (default: Infinity)
+   * @param {function} props.onConnect protocol "onConnect" handler
+   * @param {function} props.onDisconnect protocol "onDisconnect" handler
+   * @param {Array<string>} props.multicodecs protocol multicodecs
+   * @param {Registrar} registrar
+   * @constructor
+   */
+  constructor ({
+    min = 0,
+    max = Infinity,
+    onConnect,
+    onDisconnect,
+    multicodecs,
+    registrar
+  }) {
+    this.multicodecs = multicodecs
+    this.registrar = registrar
+    this.min = min
+    this.max = max
+    this.peers = new Map()
+
+    // Handlers
+    this._onConnect = onConnect
+    this._onDisconnect = onDisconnect
+
+    this._onProtocolChange = this._onProtocolChange.bind(this)
+
+    // Set by the registrar
+    this._peerStore = null
```

Comment:
After thinking about this more we should just reference the peerstore from the registrar, `this.registrar.peerStore`.